daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/RunnerFactory.scala

```diff
@@ -91,75 +98,127 @@ object Runner {
  * test can be run in parallel.
  *
  * Note however, that each thread will get its own copy of the DFDLTestSuite
+ *
+ * source is either a scala.xml.Elem, generally used for unit testing or a
+ * String. If the string starts with a / is it treated is if it were resoucrce
+ * on the classpath. Otherwise it is treated as if it were a URI.
  */
-class Runner private (elem: scala.xml.Elem, dir: String, file: String,
+class Runner private (
+  source: Either[scala.xml.Elem, String],
   validateTDMLFile: Boolean = true,
   validateDFDLSchemas: Boolean = true,
   compileAllTopLevel: Boolean = false,
   defaultRoundTripDefault: RoundTrip = Runner.defaultRoundTripDefaultDefault,
   defaultValidationDefault: String = Runner.defaultValidationDefaultDefault,
   defaultImplementationsDefault: Seq[String] = Runner.defaultImplementationsDefaultDefault) {
 
-  /*
-   * these constructors are for use by Java programs
+
+  /**
+   * Create a runner for the path. This constructor requires this path be on
+   * the classpath. A forward slash is prepended to ensure this path aways
+   * works as a classpath resource.
    */
-  def this(dir: String, file:String) =
-    this(null, dir, file)
+  def this(path: String) =
+    this(
+      if (path.startsWith("/")) Right(path)
+      else Right("/" + path)
+    )
 
+  /**
+   * Create a runner for the dir + file combination. This constructor requires
+   * this combined path be on the classpath. A forward slash is prepended to
```

Comment:
This comment is wrong. It's a cut paste from prior method doc. This forward slash is appended if needed, but it doesn't have to do with classpath. Just has to do with making sure the dir and filename are separated by a "/".

Reply:
It actually ends up calling the `path: String` constructor which will prepend the slash if needed. When using strings with the constructors (either dir + file or path), we assume that it's a resource and so the prepended slash is required.  I'll update the comment to make it more clear.